### PR TITLE
fix: placement_min_length_ratio handling for spaced point sprites

### PR DESCRIPTION
**Pull request by hjanetzek**
https://github.com/tangrams/tangram-es/issues/1313

tested with:
```
        arrows:
            filter: { oneway: yes, shield_text: false, $zoom: { min: 8 }, name: "Eastern Parkway Service Rd." }
            draw:
                icons:
                    flat: true
                    priority: 0
                    sprite: arrow
                    size: [[17, 18px], [18, 20px], [20, 32px]]
                    placement: spaced
                    #placement_spacing: [[17, 70px], [20, 175px]]
                    placement_spacing: 10px
                    angle: auto
                points:
                    priority: 0
                    collide: false
                    color: red
                    size: 5px
                    placement: vertex
```

### CODE REVIEW

File: core/src/util/lineSampler.h [hunk 1]
```diff
@@ -202,19 +180,25 @@ struct LineSampler {
...
+    }
+    float lengthToPrevSegment() {
+        if (m_curPoint >= m_points.size()) { return 0; }
+        return m_curAdvance - m_points[m_curPoint].z;
     }
 
     LineSampler(Points _points) : m_points(_points) { }
     LineSampler() { }
+
+    auto& points() { return m_points; }
```
Comment (tallytalwar) [inline core/src/util/lineSampler.h]: Doesn't seemed to be used in the current implementation, can we remove this?

**Comment by hjanetzek** [comment]
The dots are placed on each vertex to indicate line segment start and end. Looks right to me :)

File: core/src/util/lineSampler.h (hunk 1 above)
Comment (hjanetzek) [inline core/src/util/lineSampler.h]: hmm yeah can be removed

**Comment by tallytalwar** [comment]
Ahaa.. thats right. I was reading it incorrectly. Thanks for the clarification.

**Pull request merged**